src/shopilingus/Shop.java

```diff
@@ -112,7 +157,7 @@ public Product getProduct(int productId) {
 
     public void deleteProduct(int productId) {
         for (ProductDetails productDetails : products) {
-            if (productDetails.product.id == productId) {
+            if (productDetails.getProduct().getId() == productId) {
```

Comment:
Modifying (adding/removing elements) arrays/lists during iteration over them may be a serious error. Please move `this.products.remove` outside of for